ls1intum/Artemis
# `General`: Do not show certain buttons for tutors in lecture overview and course management

### Checklist
#### General


#### Client

### Motivation and Context


Fixes #7422 and also some other occurrences I noticed when fixing this issue.

### Description

Fixes authorities and hides the following buttons from tutors:
- Manage button in the lecture view inside a course (the one from #7422)
- "Edit"/"Attachment"/"Units" buttons in the manage page of a lecture
- Scores button in the course management tab bar
- In the general course management the "Tutorials" button in the course card, if there is no tutorial configuration yet

Also shows the tutorials button in the course management tab bar of a specific course so tutors have access to their tutor groups and can sign up students. This is only possible after an instructor has configured the tutorial groups.

### Steps for Testing

Prerequisites:
- 1 Tutor
- 1 Editor/Instruktor

1. Check that the buttons are successfully hidden from tutors in a course without a tutorial group configuration
2. Check that instructors see all buttons
3. Enable tutorial groups (the tutorials tab guides the instructor through this) or find a course with tutorial groups enabled
4. Check that tutors can now access tutorials in the course management
(Also see attached screenshots and links to examples on TS1)

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Screenshots


The hidden Buttons:

Manage button, e.g. here: https://artemis-test1.artemis.in.tum.de/courses/118/lectures/215
<img width="572" alt="Bildschirmfoto 2023-10-19 um 17 04 08" src="https://github.com/ls1intum/Artemis/assets/38322605/39f2d2f3-4aea-4b9a-9e57-9f4748069a55">

"Edit"/"Attachment"/"Units" buttons, e.g. here: https://artemis-test1.artemis.in.tum.de/course-management/118/lectures/215
<img width="572" alt="Bildschirmfoto 2023-10-19 um 17 04 19" src="https://github.com/ls1intum/Artemis/assets/38322605/f8a01080-2221-43e2-b41c-04d810f6c0c9">

Scores button and tutorials button not visible to tutors, e.g. here: https://artemis-test1.artemis.in.tum.de/course-management/118
<img width="680" alt="Bildschirmfoto 2023-10-21 um 11 33 25" src="https://github.com/ls1intum/Artemis/assets/38322605/afbe1ca0-6556-4d5c-9e26-59237bae9661">

Tutorials button visible in tab bar, e.g. here: https://artemis-test1.artemis.in.tum.de/course-management/118
<img width="773" alt="Bildschirmfoto 2023-10-21 um 11 34 59" src="https://github.com/ls1intum/Artemis/assets/38322605/6980dc96-63b3-4c4a-b6d6-79cb12f7e55f">

Tutorials button visible in course management card, e.g. here: https://artemis-test1.artemis.in.tum.de/course-management under the test course section
<img width="1719" alt="Bildschirmfoto 2023-10-21 um 11 35 37" src="https://github.com/ls1intum/Artemis/assets/38322605/6670da20-9583-4f27-95aa-c1ef00019cd1">
